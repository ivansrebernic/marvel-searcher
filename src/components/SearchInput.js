import React from 'react'
import styled from 'styled-components'



const SearchInputStyled = styled.input`

    width:100%;
    height:2.5rem;
    border: none;
    align-self: center;
    font-size:24px;
    &::placeholder{
        color:rgb(192,192,192);
    }
    &:focus, textarea:focus, select:focus{

    }
`




function SearchInput(props) {


    return (
        <SearchInputStyled type="text" value={props.children} onFocus={props.handleFocus} onBlur={props.handleFocus} onChange={props.handleQuery} placeholder="Search" />
    )


}

export default SearchInput