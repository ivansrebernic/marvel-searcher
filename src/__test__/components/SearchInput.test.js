import React from 'react'
import { shallow } from 'enzyme'
import SearchInput from '../../components/SearchInput'


describe('<SearchInput>', () => {
    it('Renders correctly', () => {
        const wrapper = shallow(<SearchInput></SearchInput>)
        expect(wrapper).toMatchSnapshot()
    });
})