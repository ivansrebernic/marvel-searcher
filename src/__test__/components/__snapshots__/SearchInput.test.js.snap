// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput> Renders correctly 1`] = `
<styled.input
  placeholder="Search"
  type="text"
/>
`;
